File: sdks/php/src/Webhooks/Requests/WebhooksControllerPaystackRequest.php
<?php

namespace NovaBilling\Webhooks\Requests;

use NovaBilling\Core\Json\JsonSerializableType;

class WebhooksControllerPaystackRequest extends JsonSerializableType
{
    /**
     * @var string $paystackSignature Paystack HMAC-SHA512 signature
     */
    public string $paystackSignature;

    /**
     * @param array{
     *   paystackSignature: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->paystackSignature = $values['paystackSignature'];
    }
}

File: sdks/csharp/src/NovaBilling/Webhooks/Requests/WebhooksControllerPaystackRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record WebhooksControllerPaystackRequest
{
    /// <summary>
    /// Paystack HMAC-SHA512 signature
    /// </summary>
    [JsonIgnore]
    public required string PaystackSignature { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
